batch2/batch/worker.py

```diff
@@ -504,31 +518,53 @@ def id(self):
     async def get_log(self):
         return {name: await c.get_log() for name, c in self.containers.items()}
 
-    async def delete(self):
+    async def delete(self, worker):
         log.info(f'deleting {self}')
         self.deleted = True
         for _, c in self.containers.items():
             await c.delete()
 
+        if self.start_time and not self.end_time:
+            await self.mark_cancelled(worker, time.time())
+
+    async def mark_started(self, worker, start_time):
```

Comment:
If you take my other suggestions, then I think set timing can become set start because it is only used to set the start.